BaseTools/Plugin/OverrideValidation/OverrideValidation.py

```diff
@@ -692,7 +692,10 @@ def path_parse():
         # Generate and print the override for pasting into the file.
         # Use absolute module path to find package path
         pkg_path = pathtool.GetContainingPackage(Paths.TargetPath)
-        rel_path = Paths.TargetPath[Paths.TargetPath.find(pkg_path):]
+        if pkg_path is not None:
+            rel_path = Paths.TargetPath[Paths.TargetPath.find(pkg_path):]
+        else:
+            rel_path = os.path.relpath (Paths.TargetPath, Paths.WorkSpace)

```

Comment:
I think the bigger issue here is that `BaseTools` is technically not an edk2 package.

However, it is still nested at the first level within a package path (`MU_BASECORE`).

`GetEdk2RelativePathFromAbsolutePath()` considers the given directory's relationship to package paths so it can still return the old value (`BaseTools/Conf/tools_def.template`).


```suggestion
        if pkg_path is not None:
            rel_path = Paths.TargetPath[Paths.TargetPath.find(pkg_path):]
        else:
            rel_path = pathtool.GetEdk2RelativePathFromAbsolutePath(Paths.TargetPath)

            if not rel_path:
                print(f"{Paths.TargetPath} is invalid for this workspace.")
                return
```

I don't see why OverrideValidation would process a file not relative to the workspace or a package path so I added the if statement.

Reply:
I agree with you about basetools. But we also had this working prior to an update to edk2-pytool-library, so we have a regression.

I'll test the suggested changes and verify that it's working.

Reply:
Pushed suggested changed to PR, thanks for the help.

Reply:
No problem. As Joey mentioned, we made a change to only have actual edk2 packages get returned from the function. That strictness helps more rigorously align with what the expected definition of a package is.